### DISCUSSION TITLE
Cannot Obtain Error Code Mobile URL String in either Safari, Firefox or Chrome.

### DISCUSSION THREAD

**Discussion by DocIOTUpdate** [homebridge-simplisafe3/homebridge-simplisafe3#405] [Q&A] [answered]
All, 

I have not dissabled this plugin as it has stoped working in recent updates - if you login as you request using the button on the config then it will ask for USERNAME/PASSWORD and then yoru mobile device text message code. Once this is done and you click the button nothing happens. However, if you go to https://auth.simplisafe.com/ directly and login then you get the dashboard. I have tried everthing and cannot get the mythical URL. Haw SimpliSafe not blocked the Community using this authentication?

-------------------- 

[10/8/2023, 9:18:18 PM] [Home Alarm] SimpliSafe login failed with error: {
  message: 'Request failed with status code 400',
  name: 'Error',
  description: undefined,
  number: undefined,
  fileName: undefined,
  lineNumber: undefined,
  columnNumber: undefined,
  stack: 'Error: Request failed with status code 400\n' +
    '    at createError (/usr/local/lib/node_modules/homebridge-simplisafe3/node_modules/axios/lib/core/createError.js:16:15)\n' +
    '    at settle (/usr/local/lib/node_modules/homebridge-simplisafe3/node_modules/axios/lib/core/settle.js:17:12)\n' +
    '    at IncomingMessage.handleStreamEnd (/usr/local/lib/node_modules/homebridge-simplisafe3/node_modules/axios/lib/adapters/http.js:322:11)\n' +
    '    at IncomingMessage.emit (node:events:529:35)\n' +
    '    at endReadableNT (node:internal/streams/readable:1368:12)\n' +
    '    at processTicksAndRejections (node:internal/process/task_queues:82:21)',
  config: {
    transitional: {
      silentJSONParsing: true,
      forcedJSONParsing: true,
      clarifyTimeoutError: false
    },
    adapter: [Function: httpAdapter],
    transformRequest: [ [Function: transformRequest] ],
    transformResponse: [ [Function: transformResponse] ],
    timeout: 0,
    xsrfCookieName: 'XSRF-TOKEN',
    xsrfHeaderName: 'X-XSRF-TOKEN',
    maxContentLength: -1,
    maxBodyLength: -1,
    validateStatus: [Function: validateStatus],
    headers: {
      Accept: 'application/json, text/plain, */*',
      'Content-Type': 'application/json',
      Host: 'auth.simplisafe.com',
      'Auth0-Client': 'eyJ2ZXJzaW9uIjoiMi4zLjIiLCJuYW1lIjoiQXV0aDAuc3dpZnQiLCJlbnYiOnsic3dpZnQiOiI1LngiLCJpT1MiOiIxNi4zIn19',
      'User-Agent': 'axios/0.26.1',
      'Content-Length': 77
    },
    baseURL: 'https://auth.simplisafe.com/oauth',
    method: 'post',
    url: '/token',
    data: '{"grant_type":"refresh_token","client_id":"42aBZ5lYrVW12jfOuu3CQROitwxg9sN5"}',
    'axios-retry': { retryCount: 0, lastRequestTime: 1696796298514 }
  },
  code: undefined,
  status: 400
}
[10/8/2023, 9:18:18 PM] [Home Alarm] See the plugin README for more information on authenticating with SimpliSafe.
[10/8/2023, 9:18:18 PM] [Home Alarm] Initial accessories refresh failed with error: Error: Not authenticated with SimpliSafe.
    at /usr/local/lib/node_modules/homebridge-simplisafe3/index.js:101:56
    at processTicksAndRejections (node:internal/process/task_queues:95:5)

**Comment by shamoon** [comment, marked as answer]
In general I find chrome to be easiest, I think Safari still just doesnt work. Are you looking for the URL in the browser console? Thats where it should be

**Comment by DocIOTUpdate** [comment]
I opened View > Developer > Javascript Console and this opened a separate window that then listed the correct 'mobile' URL which I needed. Thank you.

Might I recommend that in the Plugin Readme you include alongside not to use Safari, that the recomended approach is to use Google Chrome with the Javascript Console open before you start! Just an idea!

**Reply by shamoon** [reply]
I think it does? From the readme:

> You will be redirected to a page requesting you to approve the login, either via email or 2FA.
> After approval, most browsers will not redirect you to the final URL (begins with com.SimpliSafe.mobile://) but will show an error in the console from which you will need to copy & paste the URL back into the Homebridge settings UI. See notes below about using certain browsers / platforms.
> ℹ️ Many browsers (e.g. Chrome) will not redirect you and will only show an error in the Console (e.g. View > Developer Tools > Javascript Console) and you will have to copy and paste the URL from the error message.
> Safari v15.1+ neither displays the URL in the console nor visually in the URL bar and thus is not recommended for this process.

**Reply by DocIOTUpdate** [reply]
Try this: 

- Using Chrome (prior to login) open View > Developer Tools > Javascript Console keeping this window open. 
- Progress with the login of SimpiSafe where at the end you will be redirected to a page requesting you to approve the login, either via email or 2FA text message typically. 
- After approval, Chrome will not redirect you to the final URL (begins with com.SimpliSafe.mobile://). Look at the red error in the console from which you will need to then copy & paste the URL only back into your Homebridge plugin settings UI. 
- Once you do this reboot the Homebridge server after saving your settings and viola it works!   

Note: many browsers (e.g. Chrome) will not redirect you and will only show an error in the Console (e.g. View > Developer Tools > Javascript Console) and you will have to copy and paste the URL from the error message. Safari v15.1+ neither displays the URL in the console nor visually in the URL bar and thus is not recommended for this process. If you wish to login to the Dashboard outside of the Plugin then use the URL https://auth.simplisafe.com/.
